queries.sql

```diff
@@ -7,4 +7,37 @@ SELECT date_of_birth from animals WHERE name IN ('Agumon', 'Pikachu');
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered;
 SELECT * FROM animals WHERE name NOT IN ('Gabumon');
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+
+BEGIN TRANSACTION;
+UPDATE animals SET species='unspecified';
+SELECT *FROM animals;
+ROLLBACK;
+SELECT *FROM animals;
+
+BEGIN TRANSACTION;
+UPDATE animals SET species='digimon' WHERE name LIKE '%mon';
+SELECT *FROM animals;
+UPDATE animals SET species='pokemon' where species IS NULL;
+SELECT *FROM animals;
+
+BEGIN TRANSACTION;
+DELETE FROM animals;
+ROLLBACK;
+SELECT *FROM animals;
+
+ BEGIN TRANSACTION;
+ DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+ SAVEPOINT del_1;
+ UPDATE animals SET weight_kg = weight_kg * -1;
+ ROLLBACK TO del_1;
+ UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+ COMMIT;
+
+ SELECT COUNT(*) FROM animals;
+ SELECT COUNT(*) FROM animals WHERE escape_attempts > 0;
```

Comment:
- [ ] Ouch! it seems you've misinterpreted the question for this query as it asks "How many animals have never tried to escape?" but you're retrieving the animals that have tried to escape, so please reevaluate it.